documentation/asciidoc/stories/assembly_connecting_clients.adoc

```diff
@@ -0,0 +1,30 @@
+ifdef::context[:parent-context: {context}]
+[id='connecting_clients']
+:context: clients
+= Establishing Remote Client Connections
+Connect to {brandname} clusters from the {brandname} Console, Command Line Interface (CLI), and remote clients.
+Remote client connections let you create caches from valid configuration in XML or JSON format, operate on data, and perform administrative operations such as retrieving statistics and monitoring cluster health.
+
+include::{topics}/con_client_connection_details.adoc[leveloffset=+1]
+
+//Console
+include::{topics}/proc_connecting_console.adoc[leveloffset=+1]
+
+//CLI
+include::{topics}/con_cli_connections.adoc[leveloffset=+1]
+include::{topics}/proc_creating_caches_cli.adoc[leveloffset=+2]
+include::{topics}/proc_creating_caches_batch.adoc[leveloffset=+2]
+
+//Hot Rod
+include::{topics}/con_hotrod_clients.adoc[leveloffset=+1]
+include::{topics}/ref_hotrod_configuration_builder.adoc[leveloffset=+2]
+include::{topics}/ref_hotrod_properties.adoc[leveloffset=+2]
+include::{topics}/proc_creating_caches_hotrod.adoc[leveloffset=+2]
```

Comment:
@Crumby Perhaps this is going in the opposite direction of the "compact" documentation approach we took for clients in the previous versions. For downstream integrations I can also see the need to provide more detail around Hot Rod client configuration. I think it will be a big benefit for other teams if we can give more recommended examples. So while this PR introduces a lot more doc it's probably worth the effort to tell users the one best way.

Reply:
@rigazilla and @ryanemerson From a previous discussion around dogfooding, what do you guys think about taking the approach where we recommend adding cache configuration to XML|JSON files and creating them via CLI or Console? 

There may be instances where users need to create caches via Hot Rod but would you say it's approach to discourage doing so in docs? Mainly because passing long XML as a string is brutal.

Reply:
I think the recommended approach to creating caches should be:

1. Cache CR (current shortcomings are being addressed)
2. CLI (Batch CR adds direct integration with the operator, PR coming this week).
3. Console
4. HotRod

Reply:
What about REST?

Reply:
@Crumby This might be misguided but I feel like REST is kind of a "generic" option. In fairness it would make things more complete if we mentioned the REST API and perhaps gave an example similar to the console. 

If you're using an HTTP client, such as curl, you can access the REST interface at `$SERVICE_HOSTNAME:$PORT/rest/v2` and then we point off to the API docs.

At the same time I think that the REST API should be more of a "backend" thing for k8s/OpenShift. As you know, both the console and CLI are the frontends to the REST interface. Maybe we should just advocate those tools and steer users away from every possible option. I don't know if there are any scenarios for integration that use REST anyway.

Just seems to me that we should be providing more "opinionated" usage examples and including REST opens another possibility that takes us away from that approach.

Reply:
@oraNod I'm ok with not mentioning REST at all. There is whole section on REST in another doc. The only thing important here is to understand how to expose the cluster outside and the rest for the REST is the same

Reply:
> If you're using an HTTP client, such as curl, you can access the REST interface at `$SERVICE_HOSTNAME:$PORT/rest/v2` and then we point off to the API docs.

+1 to not going into usage details, but I think adding this comment somewhere would be good so that the user is at least aware it's an option.